src/converter/registry.cpp

```diff
@@ -96,9 +98,13 @@ namespace
   template< typename T >
   void delete_node( T* node )
   {
-      if( !!node && !!node->next )
+      if( node && node->next )
+      {
           delete_node( node->next );
-      delete node;
+          delete node;
+      }
+      else if( node )
```

Comment:
You are checking if `node` is not null two times instead of one. And the `delete` operator is a no-op when the pointer is null, this whole thing can be simply written as:

    if (node != 0)
        delete_node(node->next);
    delete node;